### PR TITLE
feature/396-a-learner-automatically-sees-required-learning-as-they-are-required-to-complete-it-m

### PR DESCRIPTION
**Pull request by superafroman**
Most of the changes for this actually made it in with Jen's PR. This just includes a few tests and some changes to make sure dates are Dates.

### CODE REVIEW

File: src/lib/model.ts [hunk 1]
```diff
@@ -61,39 +61,50 @@ export class Course {
 	}
 
 	nextRequiredBy() {
-		if (!this.requiredBy || !this.frequency) {
+		const [last, next] = this._currentRecurrancePeriod()
```
Comment (tav) [inline src/lib/model.ts]: `Recurrence` ?

File: test/headless/src/extension/xapi.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import axios from 'axios'
+
+const {XAPI_URL = '', XAPI_PASS = '', XAPI_USER = ''} = process.env
```
Comment (tav) [inline test/headless/src/extension/xapi.ts]: Put these in the new `test/config` module?

File: test/headless/src/page/learningPlan.ts [hunk 3]
```diff
@@ -4,4 +4,7 @@ export const selectors: Record<string, string> = {
 	learningHeader: '#content > div.main-content > div > div > h2.heading-medium',
 	requiredLearingSection: '#content > div.main-content > div > h3',
 	learningPlanSection: '#content > div.main-content > div > h2.heading-small',
+    counterFraudCourse: 'a[href="/courses/0x390"]',
```
Comment (tav) [inline test/headless/src/page/learningPlan.ts]: Inconsistent spacing?

File: test/headless/src/test/recurringCourses.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
+import * as puppeteer from 'puppeteer'
+import * as helper from 'extension/helper'
+import {wrappedBeforeAll, wrappedAfterAll} from 'extension/testsetup'
+import {createUser, updateUser} from 'extension/user'
+import * as xapi from 'extension/xapi'
+import {loginToCsl} from 'page/login'
+import {selectors} from 'page/learningPlan'
+
+const {
```
Comment (tav) [inline test/headless/src/test/recurringCourses.ts]: Use the new `test/config` module here too?

File: src/lib/model.ts (hunk 1 above)
Comment (superafroman) [reply src/lib/model.ts]: Couldn't think of anything better - the last and next date the course occurs again (or recurs).

File: test/headless/src/extension/xapi.ts (hunk 2 above)
Comment (superafroman) [reply test/headless/src/extension/xapi.ts]: Done

File: test/headless/src/page/learningPlan.ts (hunk 3 above)
Comment (superafroman) [reply test/headless/src/page/learningPlan.ts]: Done

File: src/lib/model.ts (hunk 1 above)
Comment (tav) [reply src/lib/model.ts]: Sorry, was suggesting it might be a misspelling — `a` instead of `e`?

**Pull request merged**